content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,230 @@
+---
+slug: 35
+title: 35/WAKU2-NOISE
+name: Noise Protocols for Waku2 Payload Encryption
+status: raw
+editor: Giuseppe <giuseppe@status.im>
+contributors: 
+---
+
+This specification describes how Waku2 messages can be encrypted
+in order to achieve confidentiality, authenticity, and integrity 
+as well as some form of identity-hiding on communicating parties.
+
+Specifically, it describes how encryption keys can be exchanged using the [Noise protocol Framework](http://www.noiseprotocol.org/noise.html) 
+to allow authenticated encryption/decryption in [10/WAKU2](/spec/10) with [14/WAKU-MESSAGE version 2](/spec/14/#version2). 
+This ultimately allows Waku2 users to instantiate end-to-end encrypted communication channels.
+
+
+This specification is an addition to [26/WAKU-PAYLOAD](/spec/26).
+
+## Design requirements
+
+- *Confidentiality*: 
+The adversary should not be able to learn what data is being sent from one Waku2 endpoint to one or several other Waku2 endpoints. 
+In particular, forward secrecy should be provided on exchanged data. 
+- *Authenticity*: The adversary should not be able to cause a Waku2 endpoint to accept messages coming from an endpoint different than their original senders.
+- *Integrity*: The adversary should not be able to cause a Waku2 endpoint to accept data that has been tampered with.
+- *Identity-hiding*: A passive adversary should not be able to link encrypted messages to their corresponding senders and recipients.
+
+
+## Supported Cryptographic Protocols
+
+### Noise Protocols
+
+Two parties executing a Noise protocol exchange one or more *handshake messages* and/or *transport messages*.
+A Noise protocol consists of one or more Noise handshakes. 
+During a Noise handshake, two parties exchange multiple handshake messages. 
+A handshake message contains *ephemeral keys* and/or *static keys* from one of the parties. 
+These public keys are used to perform a protocol-dependent sequence of Diffie-Hellman operations, 
+whose results are all hashed into a shared secret key. 
+After a handshake is complete, each party will then use the derived shared secret key to send and receive authenticated encrypted transport messages. 
+We refer to [Noise protocol framework specifications](http://www.noiseprotocol.org/noise.html#processing-rules) for the full details on how parties shared secret key is derived from each exchanged message.
+
+Four Noise handshakes are currently supported: `K1K1`, `XK1`, `XX`, `XXpsk0`
+(their description can be found [here](https://forum.vac.dev/t/noise-handshakes-as-key-exchange-mechanism-for-waku2/130)). 
```

Comment:
I wouldn't refer to forum post in spec probably - either Noise spec or summarize in Appendix in this spec

Reply:
Addressed in https://github.com/vacp2p/rfc/pull/496/commits/8becd7bf7c5bf8575877b8cdc4444a4b9e5d1463